### ISSUE TITLE
on object view page, make work title (up to copy designation) in banner clickable to open up an overlay showing the workinfo text and copy rows

### ISSUE THREAD

**Issue by foxm** [blakearchive/archive#250]

**Comment by foxm** [comment]
@queryluke i started playing around with this in branch overlay, but i couldn't get it to pass the overlay content from work-title to main. notice 
`<div class="overlay-content">
        test
 </div>`
in main. that should be dynamically loaded with an ng-click of the work title in the banner or the word "Copy." Clicking the work title should bring up in the overlay the text of the info page. Clicking "Copy" should bring up in the overlay the grid of copies."

In workTitle.html I started to try to implement this for works with titles and copy designations (the last ahref in workTitle.html), but I couldn't get it to work. I tried to set the overlay content and then use that in main.html to no avail. 

When you get a chance can you look at this? No rush. I'll keep working on it.

**Comment by foxm** [comment]
@queryluke made some progress on this. now the workinfo is showing, but i'd like to get the copies showing in rows as they do on the work info page. i tried copying the code from the work controller, but that didn't seem to work. is there a way to inherit the workCtrl? not sure if that's the right way to go about this.

**Comment by foxm** [comment]
at the moment in branch overlay, you have to click on the title twice to set the work and see the work info and copies in the overlay. i tried setting the work info initially automatically, but that crashes the site.

**Comment by queryluke** [comment]
@foxm I took a quick look at this. First and foremost, I would hold off on anymore development. If we're going to replicate the works page in an overlay, we need to break out the works page pieces (description, copies, and even title) into their own separate directives. This way we don't need to repeat the logic between the work controller and the workTitle directive.

I'm working on the comparison and zoom bugs this week, but I can circle back around to this when I'm done.

**Comment by foxm** [comment]
@queryluke thanks, luke. that's what i was thinking. repeating the logic was causing all kinds of problems that were crashing the site. (or, at least the way i was using that logic was crashing the site.)

**Comment by foxm** [comment]
@queryluke luke, i got rid of the redundancy of logic. the overlay works as it should if you reach a work through the normal channel; that is, if you go from the homepage to the work page to the object view page or from the homepage to the object view page. but if you reload the object view page, the overlay becomes incomplete; it shows only the info text and not the copy grid. i think it's just a matter of setting the selected work properly. i add a setWork() function to the workController, but for some reason that doesn't pick up the copies.

**Comment by queryluke** [comment]
@foxm 
I think i've got the overly branch working. have a look and let me know if you come across any bugs

**Comment by foxm** [comment]
@queryluke nice job, luke. one thing: the home page isn't populating the featured works. i tried to fix it, but i'm confused as to how the api (/api/featured_works) works.

**Comment by foxm** [comment]
@queryluke also, there are some issues now with the titling. going from compare view to a work page leaves "(Selected)" up. and the letters in object view should look like this: Letters: To...," where 'Letters' is linked to the overlay.

**Comment by foxm** [comment]
fixed titling for letters and got rid of (Selected) problem

**Comment by foxm** [comment]
@queryluke luke, there seems to be a bug with Letters. i added some logic to deal with the titling of letters, but when you go from Letters (in the object view) to another work (the work info page of another work), you get a blank title, which, by the logic i've set up, means workTitle.bds.copy.bad_id is not getting reset from 'letters'.

**Comment by foxm** [comment]
@queryluke ok, i think i fixed all the bugs

**Comment by foxm** [comment]
nope, spotted another: the textual links to other works inside the work info inside the overlay window aren't displaying or working properly.

**Comment by foxm** [comment]
fixed

**Comment by queryluke** [comment]
@foxm great. feel free to merge and close

**Comment by foxm** [comment]
@queryluke woops, i didn't fix the homepage issue. for some reason the images aren't showing on the homepage

**Comment by queryluke** [comment]
@foxm 
Figured it out, you need to run ```python homepageimport.py``` via terminal. Need to be in the ```blakearchive/``` dir.

**Comment by foxm** [comment]
@queryluke 

Hmm, ok, just tried that and it didn't seem to work. I also got the following warnings:

/Users/michaelfox/Sites/blake/lib/python2.7/site-packages/sqlalchemy/sql/sqltypes.py:185: SAWarning: Unicode type received non-unicode bind param value 'america.a.p9'. (this warning may be suppressed after 10 occurrences)
  (util.ellipses_string(value),))
/Users/michaelfox/Sites/blake/lib/python2.7/site-packages/sqlalchemy/sql/sqltypes.py:185: SAWarning: Unicode type received non-unicode bind param value 'america.a.illbk.09'. (this warning may be suppressed after 10 occurrences)
  (util.ellipses_string(value),))
/Users/michaelfox/Sites/blake/lib/python2.7/site-packages/sqlalchemy/sql/sqltypes.py:185: SAWarning: Unicode type received non-unicode bind param value 'America a Prophecy'. (this warning may be suppressed after 10 occurrences)
  (util.ellipses_string(value),))
/Users/michaelfox/Sites/blake/lib/python2.7/site-packages/sqlalchemy/sql/sqltypes.py:185: SAWarning: Unicode type received non-unicode bind param value 'america.m.p2'. (this warning may be suppressed after 10 occurrences)
  (util.ellipses_string(value),))
/Users/michaelfox/Sites/blake/lib/python2.7/site-packages/sqlalchemy/sql/sqltypes.py:185: SAWarning: Unicode type received non-unicode bind param value 'b-los.a.p1'. (this warning may be suppressed after 10 occurrences)
  (util.ellipses_string(value),))
/Users/michaelfox/Sites/blake/lib/python2.7/site-packages/sqlalchemy/sql/sqltypes.py:185: SAWarning: Unicode type received non-unicode bind param value 'The Book of Los'. (this warning may be suppressed after 10 occurrences)
  (util.ellipses_string(value),))
/Users/michaelfox/Sites/blake/lib/python2.7/site-packages/sqlalchemy/sql/sqltypes.py:185: SAWarning: Unicode type received non-unicode bind param value 'bb85.a.1.ps'. (this warning may be suppressed after 10 occurrences)
  (util.ellipses_string(value),))
/Users/michaelfox/Sites/blake/lib/python2.7/site-packages/sqlalchemy/sql/sqltypes.py:185: SAWarning: Unicode type received non-unicode bind param value 'copy A, object 1 -- "Albi...'. (this warning may be suppressed after 10 occurrences)
  (util.ellipses_string(value),))
/Users/michaelfox/Sites/blake/lib/python2.7/site-packages/sqlalchemy/sql/sqltypes.py:185: SAWarning: Unicode type received non-unicode bind param value 'Illustrations to the Book...'. (this warning may be suppressed after 10 occurrences)
  (util.ellipses_string(value),))
/Users/michaelfox/Sites/blake/lib/python2.7/site-packages/sqlalchemy/sql/sqltypes.py:185: SAWarning: Unicode type received non-unicode bind param value 'bb448.1.spb.05'. (this warning may be suppressed after 10 occurrences)

**Comment by foxm** [comment]
@queryluke ok, i merged this into master, but there's still the homepage issue. heading home for the day.

**Comment by foxm** [comment]
@queryluke i think i fixed the homepage, although i'm not sure if i broke something else in the process. i had to edit getFeaturedWorks in main.js:

function getFeaturedWorksComplete(response){
                //return BlakeWork.create(response.data.results);
                return response.data.results;
}

you can roll this back if you think it might break something else, but it seems to be only used by the homepage

**Comment by queryluke** [comment]
@foxm 
Ah, I got confused as to which branch we were talking about. I was on master (before overlay was merged) and my homepage was broken (then fixed by reimporting). I see the problem now. I had moved logic into the BlakeWork service, but the homepage works don't have the same data model as an actual work. So it makes sense NOT to pass them through the BlakeWork service (as we do with the initial search results). So yeah, there change you made is spot on for now.
